lib/cfa/augeas_parser.rb

```diff
@@ -38,59 +47,85 @@ def add(value, placer = AppendPlacer.new)
       element = placer.new_element(@tree)
       element[:key] = augeas_name
       element[:value] = value
+      element[:operation] = :add
       # FIXME: load_collection missing here
     end
 
     def delete(value)
-      key = augeas_name
-      @tree.data.reject! do |entry|
-        entry[:key] == key &&
-          if value.is_a?(Regexp)
-            value =~ entry[:value]
-          else
-            value == entry[:value]
-          end
-      end
+      to_delete, to_mark = to_remove(value)
+                           .partition { |e| e[:operation] == :add }
+      @tree.all_data.delete_if { |e| to_delete.include?(e) }
```

Comment:
no as `#all_data` do not have version with `#all_data=` so `@tree.all_data -= to_delete` don't work.